private/model/api/example.go

```diff
@@ -0,0 +1,379 @@
+	case "string":
+		return fmt.Sprintf("%s: aws.String(%q),\n", memName, v)
+	case "integer":
+		return fmt.Sprintf("%s: aws.Int64(%s),\n", memName, v)
+	case "boolean":
+		return fmt.Sprintf("%s: aws.Bool(%s),\n", memName, v)
+	default:
+		panic("Unsupported type: " + t)
+	}
+}
+
+func buildComplex(name, memName string, ref *ShapeRef, v map[string]interface{}) string {
+	shapeName, t := "", ""
+	member := ref.Shape.MemberRefs[name]
+
+	if member != nil && member.Shape != nil {
+		shapeName = ref.Shape.MemberRefs[name].Shape.ShapeName
+		t = ref.Shape.MemberRefs[name].Shape.Type
+	} else {
+		shapeName = ref.Shape.ShapeName
+		t = ref.Shape.Type
+	}
+
+	switch t {
+	case "structure":
+		return fmt.Sprintf(`%s: &%s.%s{
+				%s
+			},
+			`, memName, ref.API.PackageName(), shapeName, buildShape(ref.Shape.MemberRefs[name], v, false))
+	case "map":
+		shapeType := getNestedType(ref.Shape.MemberRefs[name].Shape.ValueRef.Shape)
+		return fmt.Sprintf(`%s: map[string]%s{
+				%s
+			},
+			`, name, shapeType, buildShape(&ref.Shape.MemberRefs[name].Shape.ValueRef, v, true))
+	}
+
+	return ""
+}
+
+// AttachExamples will create a new ExamplesDefinition from the examples file
+// and reference the API object.
+func (a *API) AttachExamples(filename string) {
+	a.Setup()
+	a.customizationPasses()
+	p := ExamplesDefinition{API: a}
+
+	f, err := os.Open(filename)
+	defer f.Close()
+	if err != nil {
+		panic(err)
+	}
+	err = json.NewDecoder(f).Decode(&p)
+	if err != nil {
+		panic(err)
+	}
+
+	p.setup()
+}
+
+func (p *ExamplesDefinition) setup() {
+	keys := p.Examples.Names()
+	for _, n := range keys {
+		examples := p.Examples[n]
+		for i, e := range examples {
+			n = p.ExportableName(n)
+			e.OperationName = n
+			e.API = p.API
+			e.Index = fmt.Sprintf("shared%02d", i)
+			e.VisitedErrors = map[string]struct{}{}
+			op := p.API.Operations[e.OperationName]
+			e.OperationName = p.ExportableName(e.OperationName)
+			e.Operation = op
+			p.Examples[n][i] = e
+		}
+	}
+
+	p.API.Examples = p.Examples
+}
+
+var exampleHeader = template.Must(template.New("exampleHeader").Parse(`
+import (
+	"fmt"
+	"bytes"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/{{ .PackageName }}"
+)
+
+var _ time.Duration
+var _ bytes.Buffer
+var _ aws.Config
+
+func parseTime(layout, value string) *time.Time {
+	t, err := time.Parse(layout, value)
+	if err != nil {
+		panic(err)
+	}
+	return &t
+}
+
+`))
+
+// ExamplesGoCode will return a code representation of the entry within the
+// examples.json file.
+func (a *API) ExamplesGoCode() string {
+	var buf bytes.Buffer
+	if err := exampleHeader.ExecuteTemplate(&buf, "exampleHeader", &a); err != nil {
+		panic(err)
+	}
+
+	code := a.Examples.GoCode()
+	if len(code) == 0 {
+		return ""
+	}
+
+	buf.WriteString(code)
+	return buf.String()
+}
+
+// TODO: In the operation docuentation where we list errors, this needs to be done
```

Comment:
Yea, that is part of the docstring issue of listing errors. I was going to submit that as a separate PR